lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsFields/OptionsFields.js

```diff
@@ -0,0 +1,153 @@
+    }
+
+    if (value.length > 100) {
+      return <FormattedMessage id="stripes-smart-components.customFields.characters.limit" values={{ limit: 100 }} />;
+    }
+
+    const optionLabels = get(fields, fieldNamePrefix).selectField.options.values;
+
+    if (isLabelDuplicated(value, optionLabels)) {
+      return <FormattedMessage id="stripes-smart-components.customFields.option.duplicate" />;
+    }
+
+    return null;
+  };
+
+  const getContentData = fields => {
+    return fields.map((_field, index) => ({ label: index, actions: index }));
+  };
+
+  return (
+    <FieldArray
+      isEqual={isEqual}
+      name={`${fieldNamePrefix}.selectField.options.values`}
+    >
+      {({ fields }) => {
+        const optionsListEmpty = !fields.length;
+
+        return (
+          <>
+            <Col xs={12}>
+              <div className={!optionsListEmpty ? css.listWrapper : null}>
+                <MultiColumnList
+                  contentData={getContentData(fields)}
+                  isEmptyMessage={null}
+                  visibleColumns={['label', 'code', 'default', 'actions']}
+                  columnMapping={{
+                    label:  (
+                      <Label tagName="span" required className={css.optionLabel}>
+                        <FormattedMessage id="stripes-smart-components.customFields.dropdown.label" />
+                      </Label>
+                    ),
+                    code: <FormattedMessage id="stripes-smart-components.customFields.dropdown.code" />,
+                    default: <FormattedMessage id="stripes-smart-components.customFields.dropdown.default" />,
+                    actions: null,
+                  }}
+                  formatter={{
+                    label: ({ rowIndex }) => (
+                      <Field
+                        className={css.optionInput}
+                        component={TextField}
+                        name={`${fieldNamePrefix}.selectField.options.values[${rowIndex}]`}
+                        marginBottom0
+                        required
+                        validate={labelValidation}
+                        validateFields={[]}
+                      />
+                    ),
+                    code: () => <TextField marginBottom0 disabled />,
+                    default: () => (
+                      <Field
+                        component={RadioButton}
+                        name={`${fieldNamePrefix}.selectField.defaults`}
+                        inline
+                        marginBottom0
+                        disabled
+                      />
+                    ),
+                    actions: ({ rowIndex }) => (
+                      <IconButton
+                        icon="trash"
+                        onClick={() => {
+                          setOptionToDelete(fields.value[rowIndex]);
+                          setModalState(true);
+                        }}
```

Comment:
I think opening or closing the deletion confirmation modal should happen as the result of changes to `optionToDelete` state, like this:
```js
useEffect(() => {
  if(optionToBeDeleted) {
    setConfirmationModalOpen(true);
  } else {
    setConfirmationModalOpen(false);
  }
}, [optionToBeDeleted])
```
This is not a strong preference, but it feels more conceptually correct to me.